addons/payment_stripe/models/payment_transaction.py

```diff
@@ -278,6 +280,10 @@ def _process_notification_data(self, notification_data):
         if 'charge' in notification_data:
             self.acquirer_reference = notification_data['charge']['id']
 
+        if 'payment_intent' in notification_data:
+            payment_intent = notification_data.get('payment_intent', {})
+            self.stripe_payment_intent = payment_intent.get('id') if isinstance(payment_intent, dict) else payment_intent
```

Comment:
Why do you need to do this here? Couldn't it be done when creating the payment intent? Actually, do you need to save the payment intent id at all?

Reply:
We need the payment intent id to capture, void, and refund transaction from Stripe. I added it there because we only create a payment intent for transactions using a token. Otherwise, we create a checkout session and the payment intent id is in the checkout session. 

Indeed, We could save the payment intent id when creating the checkout session but we would need to do it for tokenized transactions and also for refund transactions. This way, the payment intent id is always saved with only one code to maintain.